### PR TITLE
fix(zod-openapi): multi-middleware complex type inference

### PR DESCRIPTION
**Pull request by askorupskyy**
Fixed the inference of complex types in `zod-openapi`.

Issue explained in detail [here](https://github.com/honojs/middleware/issues/847)

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @askorupskyy 

Looks good! One thing. Can you run `yarn changeset` at the top of this repo and push the changeset? This will be a `patch` update.

**Comment by askorupskyy** [comment]
@yusukebe done!

**Comment by askorupskyy** [comment]
I also squashed everything so should be good to go

File: .changeset/stale-hairs-wonder.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@hono/zod-openapi': minor
```
Comment (yusukebe) [inline .changeset/stale-hairs-wonder.md:2]: This should be `patch` instead of `minor` because it is a bug fix.

Comment (askorupskyy) [reply .changeset/stale-hairs-wonder.md:2]: oops, will fix

Comment (askorupskyy) [reply .changeset/stale-hairs-wonder.md]: done. the cli didn't include the `patch` option for me, only minor

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@askorupskyy 

Thank you very much! Merging now.

**Pull request merged**